airbyte-db/src/main/java/io/airbyte/db/instance/JobsDatabaseInstance.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.db.instance;
+
+import static org.jooq.impl.DSL.select;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.db.Database;
+import io.airbyte.db.Databases;
+import io.airbyte.db.ExceptionWrappingDatabase;
+import io.airbyte.db.ServerUuid;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JobsDatabaseInstance implements DatabaseInstance {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JobsDatabaseInstance.class);
+  private static final Function<Database, Boolean> IS_JOBS_DATABASE_CONNECTED = database -> {
+    try {
+      LOGGER.info("Testing jobs database connection...");
+      return database.query(ctx -> ctx.fetchExists(select().from("information_schema.tables")));
+    } catch (Exception e) {
+      return false;
+    }
+  };
+  public static final Function<Database, Boolean> IS_JOBS_DATABASE_READY = database -> {
```

Comment:
Seems to be fair amount of duplicated code across `JobsDatabaseInstance` and `ConfigDatabaseInstance`. The main difference I see is the checking for the Server Uuid.

What do you think about pulling most of this logic into the `DatabaseInstance` interface as default method for easier reuse? We could also use a base class pattern with an additional 'ready' method the JobsDBInstance can implement. I foresee more tables on the private cloud so this is something we'll be able to reuse. Wdyt?

Reply:
Sure. I was not drying it because there were only two implementations. But you are right that this can be reused by cloud. Will do.